src/modules/sndio.cpp

```diff
@@ -65,7 +67,23 @@ Sndio::Sndio(const std::string &id, const Json::Value &config)
 
     int revents = sioctl_revents(hdl_, pfds_.data());
     if (revents & POLLHUP) {
-      throw std::runtime_error("disconnected!");
+      spdlog::warn("sndio disconnected!");
+      sioctl_close(hdl_);
+      hdl_ = nullptr;
+
+      // reconnection loop
+      while (1) {
```

Comment:
Could you replace with `while (thread_.isRunning())`?

Reply:
Just did.